### PR TITLE
Remove find by username invocation from listings controller

### PR DESCRIPTION
**Pull request by lyyder**
Change Person fetch so that username and community id are used in ListingsController.

### CODE REVIEW

File: app/controllers/listings_controller.rb [hunk 1]
```diff
@@ -34,9 +34,15 @@ def index
...
-          PersonViewUtils.ensure_person_belongs_to_community!(@person, @current_community)
+
+        # Username is passed in person_id parameter for historical reasons.
+        # In the future the actual parameter name should be changed to better
+        # express its purpose. However this is a large change as there are quite a few
+        # resources nested under the people resource.
+        username = params[:person_id]
+
+        if request.xhr? && username # AJAX request to load on person's listings for profile view
+          @person = Person.find_by_username_and_community_id!(username, @current_community.id)
```
Comment (sktoiva) [inline app/controllers/listings_controller.rb]: We should already start moving away from the magic dynamic `find_by_*` methods and instead use the format that Rails 5 will enforce:
`Person.find_by!(username: username, community_id: @current_community.id)`

Comment (lyyder) [inline app/controllers/listings_controller.rb]: In general yes.

In this case we are planning to deploy this change before the Hydra data migrations. So actually it's not a magic dynamic method but an implemented one from ``Person.rb`` :
````
  def self.find_by_username_and_community_id(username, community_id)
    Person
      .joins(:community_memberships)
      .where(username: username, community_memberships: { community_id: community_id })
      .first
  end

  def self.find_by_username_and_community_id!(username, community_id)
    person = Person.find_by_username_and_community_id(username, community_id)

    if person
      person
    else
      raise ActiveRecord::RecordNotFound.new("Can not find person with username #{username} and community id #{community_id}")
    end
  end
````

Comment (sktoiva) [inline app/controllers/listings_controller.rb]: Ah, ok. Somehow it feels bad that there's a similarly named dynamic method as I already did the mistake in assuming that the method refers to that. But I don't have a strong opinion on should this be fixed by renaming the methods somehow or leave it as is.

Comment (lyyder) [inline app/controllers/listings_controller.rb]: Plan is to remove the method once we run the migrations and introduce the ``community_id`` field to person. Or at latest when the ``community_memberships`` table is dropped.

**Pull request merged**